state-chain/runtime/src/lib.rs

```diff
@@ -1427,9 +1427,15 @@ impl_runtime_apis! {
 
 		fn cf_scheduled_swaps(base_asset: Asset, _quote_asset: Asset) -> Vec<ScheduledSwap> {
 
+			let current_block = System::block_number();
+
 			pallet_cf_swapping::SwapQueue::<Runtime>::iter().flat_map(|(block, swaps_for_block)| {
+
+				// In case `block` has already passed, the swaps will be re-tried at the next block:
+				let execute_at = if block > current_block { block } else { current_block.saturating_add(1) };
```

Comment:
I think this will should be `block >= current_block`, because this code is run at the end of the block N, but I assume current_block will still have the value N. But those swaps that remain in the SwapQueue at N, will be executed at N + 1.